Jason Rowsell

jasonrowsell opened the pull request:
Features:
- Sign-in (ERB)
- Choosing weapon (ERB)
- Play against AI (ERB to_json, JavaScript)
- Animation onClick (CSS keyframes, JavaScript)
- Play button disabled onClick (CSS, JavaScript)
- 'Play again' button appended animationEnd (jQuery)
- Result printed animationEnd (ERB to_json)
- Score is calculated and stored (ERB to_json)

To do:
- Implement feature tests that account for waiting-time for animationEnd (after onClick)
- Animation effect when hovering over weapon buttons (glow effect?)
- Unit test with stub to test for point increase after playing a game
- Refactor private 'point' methods whilst having correct score-count (ERB to_json, JS)

jasonrowsell committed:
Added index erb and styled

jasonrowsell committed:
Passing sign-in test with minimum

jasonrowsell committed:
Implemented JS with erb and json

jasonrowsell committed:
Play button is deleted after animation end

jasonrowsell committed:
Play Again button links back to /choice

jasonrowsell committed:
Added randomise feature to computer class

jasonrowsell committed:
Added check_winner feature, implemented into js

jasonrowsell committed:
Added points to classes

jasonrowsell committed:
Score feature ffunctioning, needs refactoring

jasonrowsell committed:
Update README.md

jasonrowsell committed:
Updated Gemfile

jasonrowsell committed:
Updated Gemfile

jasonrowsell committed:
Updated Gemfile

jasonrowsell committed:
Update README.md

jasonrowsell committed:
Update README.md

jasonrowsell committed:
Set up coveralls

jasonrowsell committed:
Set up coveralls

noarfarber commented on README.md:
```diff
@@ -1,85 +1,83 @@
-# RPS Challenge
+# Rock, Paper, Scissors with Web Framework
```
Really like your use of the README file. Instructions are very clear and engaging.

noarfarber commented on app.rb:
```diff
@@ -0,0 +1,39 @@
+    erb "choice.html".to_sym
+  end
+
+  post '/weapon' do
```
Not sure how to implement it, but maybe you could make the controller skinnier by moving functionality somewhere else?

noarfarber reviewed (commented):
App is fun and looks great.
Nicely structured, code is clear and neat.
Good use of folders and files names.
I like that you managed to implement style using CSS and make it yours.
Well done!

dependabot-preview[bot] committed:
Upgrade to GitHub-native Dependabot

jasonrowsell committed:
Cleaning up spec tests

jasonrowsell committed:
Updated Gemfile

jasonrowsell committed:
Update README.md

jasonrowsell committed:
Added coveralls

jasonrowsell committed:
Merge branch 'master' of github.com:jasonrowsell/rps-challenge

jasonrowsell committed:
Added coveralls

jasonrowsell committed:
Update gemfile

jasonrowsell committed:
Update README.md

jasonrowsell committed:
Update README.md

jasonrowsell committed:
Update README.md

jasonrowsell committed:
Update README.md

jasonrowsell committed:
Updated gemfile

jasonrowsell committed:
Update README.md

dependabot-preview[bot] committed:
[Security] Bump addressable from 2.7.0 to 2.8.0

Bumps [addressable](https://github.com/sporkmonger/addressable) from 2.7.0 to 2.8.0. **This update includes a security fix.**
- [Release notes](https://github.com/sporkmonger/addressable/releases)
- [Changelog](https://github.com/sporkmonger/addressable/blob/main/CHANGELOG.md)
- [Commits](https://github.com/sporkmonger/addressable/compare/addressable-2.7.0...addressable-2.8.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

depfu[bot] committed:
Update addressable to version 2.8.0

dependabot-preview[bot] committed:
Bump rubocop from 0.79.0 to 1.18.4

Bumps [rubocop](https://github.com/rubocop/rubocop) from 0.79.0 to 1.18.4.
- [Release notes](https://github.com/rubocop/rubocop/releases)
- [Changelog](https://github.com/rubocop/rubocop/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rubocop/rubocop/compare/v0.79.0...v1.18.4)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
Bump puma from 5.3.2 to 5.4.0

Bumps [puma](https://github.com/puma/puma) from 5.3.2 to 5.4.0.
- [Release notes](https://github.com/puma/puma/releases)
- [Changelog](https://github.com/puma/puma/blob/master/History.md)
- [Commits](https://github.com/puma/puma/compare/v5.3.2...v5.4.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

depfu[bot] committed:
Update puma to version 5.4.0

depfu[bot] committed:
Update codecov to version 0.6.0

jasonrowsell committed:
Merge pull request #36 from jasonrowsell/depfu/update/codecov-0.6.0

Update codecov: 0.5.2 → 0.6.0 (major)

jasonrowsell committed:
Merge pull request #35 from jasonrowsell/depfu/update/puma-5.4.0

Update puma: 5.3.2 → 5.4.0 (minor)

jasonrowsell committed:
Merge pull request #34 from jasonrowsell/dependabot/bundler/puma-5.4.0

Bump puma from 5.3.2 to 5.4.0

jasonrowsell committed:
Merge pull request #33 from jasonrowsell/dependabot/bundler/rubocop-1.18.4

Bump rubocop from 0.79.0 to 1.18.4

jasonrowsell committed:
Merge pull request #32 from jasonrowsell/depfu/update/addressable-2.8.0

🚨 [security] Update addressable: 2.7.0 → 2.8.0 (minor)

jasonrowsell committed:
Merge pull request #10 from jasonrowsell/dependabot/add-v2-config-file

Upgrade to GitHub-native Dependabot

jasonrowsell committed:
Merge pull request #31 from jasonrowsell/dependabot/bundler/addressable-2.8.0

[Security] Bump addressable from 2.7.0 to 2.8.0

depfu[bot] committed:
Update puma to version 5.5.0

dependabot[bot] committed:
Bump rubocop from 1.18.4 to 1.22.1

Bumps [rubocop](https://github.com/rubocop/rubocop) from 1.18.4 to 1.22.1.
- [Release notes](https://github.com/rubocop/rubocop/releases)
- [Changelog](https://github.com/rubocop/rubocop/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rubocop/rubocop/compare/v1.18.4...v1.22.1)

---
updated-dependencies:
- dependency-name: rubocop
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

jasonrowsell committed:
Merge pull request #39 from jasonrowsell/depfu/update/puma-5.5.0

Update puma: 5.4.0 → 5.5.0 (minor)

jasonrowsell committed:
Merge pull request #42 from jasonrowsell/dependabot/bundler/rubocop-1.22.1

Bump rubocop from 1.18.4 to 1.22.1

dependabot[bot] committed:
Bump puma from 5.5.0 to 5.5.1

Bumps [puma](https://github.com/puma/puma) from 5.5.0 to 5.5.1.
- [Release notes](https://github.com/puma/puma/releases)
- [Changelog](https://github.com/puma/puma/blob/master/History.md)
- [Commits](https://github.com/puma/puma/compare/v5.5.0...v5.5.1)

---
updated-dependencies:
- dependency-name: puma
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

depfu[bot] committed:
Update nokogiri to version 1.12.5

jasonrowsell committed:
Merge pull request #44 from jasonrowsell/dependabot/bundler/puma-5.5.1

Bump puma from 5.5.0 to 5.5.1

jasonrowsell committed:
Merge pull request #40 from jasonrowsell/depfu/update/nokogiri-1.12.5

🚨 [security] Update nokogiri: 1.11.7 → 1.12.5 (minor)

depfu[bot] committed:
Update puma to version 5.5.2

jasonrowsell committed:
Update README.md

dependabot[bot] committed:
Bump capybara from 3.35.3 to 3.36.0

Bumps [capybara](https://github.com/teamcapybara/capybara) from 3.35.3 to 3.36.0.
- [Release notes](https://github.com/teamcapybara/capybara/releases)
- [Changelog](https://github.com/teamcapybara/capybara/blob/master/History.md)
- [Commits](https://github.com/teamcapybara/capybara/compare/3.35.3...3.36.0)

---
updated-dependencies:
- dependency-name: capybara
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump rubocop from 1.22.1 to 1.23.0

Bumps [rubocop](https://github.com/rubocop/rubocop) from 1.22.1 to 1.23.0.
- [Release notes](https://github.com/rubocop/rubocop/releases)
- [Changelog](https://github.com/rubocop/rubocop/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rubocop/rubocop/compare/v1.22.1...v1.23.0)

---
updated-dependencies:
- dependency-name: rubocop
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

jasonrowsell committed:
Merge pull request #45 from jasonrowsell/depfu/update/puma-5.5.2

Update puma: 5.5.1 → 5.5.2 (patch)

jasonrowsell committed:
Merge pull request #47 from jasonrowsell/dependabot/bundler/capybara-3.36.0

Bump capybara from 3.35.3 to 3.36.0

jasonrowsell committed:
Merge pull request #50 from jasonrowsell/dependabot/bundler/rubocop-1.23.0

Bump rubocop from 1.22.1 to 1.23.0
